test/units/plugins/cache/test_cache.py

```diff
@@ -86,18 +98,36 @@ def test_flush(self):
         self.cache._plugin.set('another wolf', 'another animal')
 
         # The adjudicator does't know about the new entries
-        assert len(self.cache) == 2
+        assert len(self.cache.keys()) == 2
         # But the cache itself does
-        assert len(self.cache._plugin._cache) == 3
+        assert len(self.cache._plugin.keys()) == 3
 
         # If we call flush, both the adjudicator and the cache should flush
         self.cache.flush()
-        assert len(self.cache) == 0
-        assert len(self.cache._plugin._cache) == 0
+        assert len(self.cache.keys()) == 0
+        assert len(self.cache._plugin.keys()) == 0
 
 
-class TestFactCache(unittest.TestCase):
+class TestJsonFileCache(TestCachePluginAdjudicator):
+    cache_prefix = ''
+
+    def setUp(self):
+        self.cache_dir = tempfile.mkdtemp(prefix='ansible-plugins-cache-')
+        self.cache = CachePluginAdjudicator(
+            plugin_name='jsonfile', _uri=self.cache_dir,
+            _prefix=self.cache_prefix)
+        self.cache['cache_key'] = {'key1': 'value1', 'key2': 'value2'}
+        self.cache['cache_key_2'] = {'key': 'value'}
+
+    def tearDown(self):
+        shutil.rmtree(self.cache_dir)
+
+
+class TestJsonFileCachePrefix(TestJsonFileCache):
```

Comment:
Are you planning on adding a test to make sure keys() only retrieves those with the correct prefix? That would be a great addition.

Reply:
Sure I'll try to get that added before the holidays